### PR TITLE
Remove support for previously deprecated for V1 APIs

### LINKED ISSUE #1829: Need to remove the depreciated oe_get_report_v1
The following old APIs are no longer needed because OE SDK has switched to *_v2.
oe_get_report_v1
oe_get_target_info_v1
oe_get_seal_key_by_policy_v1
oe_result_t oe_get_seal_key_v1

### PR DESCRIPTION
**Pull request by paulcallen**
Default to OE_API_VERSION=2 and remove all code referring to OE_API_VERSION=1, throwing an error if built with that option.
Fixes #1829

### CODE REVIEW

File: tests/report/common/tests.cpp [hunk 1]
```diff
@@ -682,85 +553,34 @@ void test_remote_report()
...
-            GetReport_v1(
+            GetReport_v2(
                 flags,
                 report_data,
                 report_data_size,
                 NULL,
                 0,
-                report_buffer,
-                &report_size) == expected_result);
+                &report_buffer_ptr,
```
Comment (dthaler) [inline tests/report/common/tests.cpp]: Where is the corresponding call to 
oe_free_report(report_buffer_ptr);
?  I expected to see it at line 583 before exiting the 'if' block

File: tests/sealKey/enc/enc.cpp [hunk 2]
```diff
@@ -247,37 +188,29 @@ bool TestOEGetSealKey()
...
-        OE_TEST(third_key_buffer_size == sizeof(sgx_key_t));
         OE_TEST(third_key_ptr_size == sizeof(sgx_key_t));
 
         // The seal keys should match.
-        if ((memcmp(key_buffer, second_key_buffer, sizeof(sgx_key_t)) != 0) ||
-            (memcmp(key_buffer, third_key_ptr, sizeof(sgx_key_t)) != 0) ||
-            (memcmp(key_buffer, third_key_buffer, sizeof(sgx_key_t)) != 0))
+        if ((memcmp(key_buffer_ptr, third_key_ptr, sizeof(sgx_key_t)) != 0))
         {
             return false;
```
Comment (dthaler) [inline tests/sealKey/enc/enc.cpp]: I think you need to free third_key_ptr before returning false

File: tests/sealKey/enc/enc.cpp [hunk 3]
```diff
@@ -286,17 +219,14 @@ bool TestOEGetSealKey()
...
-        }
         ret = oe_get_seal_key_v2(
-            key_info, key_info_size, &third_key_ptr, &third_key_ptr_size);
+            key_info_ptr,
+            key_info_ptr_size,
+            &third_key_ptr,
+            &third_key_ptr_size);
         if (ret != OE_INVALID_ISVSVN)
         {
             return false;
```
Comment (dthaler) [inline tests/sealKey/enc/enc.cpp]: if ret is OE_OK, then you need to free third_key_ptr before returning

File: tests/sealKey/enc/enc.cpp [hunk 4]
```diff
@@ -309,18 +239,18 @@ bool TestOEGetSealKey()
...
-        }
         ret = oe_get_seal_key_v2(
-            key_info, key_info_size, &third_key_ptr, &third_key_ptr_size);
+            key_info_ptr,
+            key_info_ptr_size,
+            &third_key_ptr,
+            &third_key_ptr_size);
         if (ret != OE_INVALID_CPUSVN)
         {
             return false;
```
Comment (dthaler) [inline tests/sealKey/enc/enc.cpp]: if ret is OE_OK then you need to free here too

File: tests/report/common/tests.cpp (hunk 1 above)
Comment (paulcallen) [reply tests/report/common/tests.cpp]: FIxed

File: tests/sealKey/enc/enc.cpp (hunk 2 above)
Comment (paulcallen) [reply tests/sealKey/enc/enc.cpp]: Fixed

File: tests/sealKey/enc/enc.cpp (hunk 3 above)
Comment (paulcallen) [reply tests/sealKey/enc/enc.cpp]: Fixed

File: tests/sealKey/enc/enc.cpp (hunk 4 above)
Comment (paulcallen) [reply tests/sealKey/enc/enc.cpp]: Fixed

**Comment by soccerGB** [comment]
bors try

**Pull request merged**